packages/alfa-rules/src/common/applicability/targets-of-pointer-events.ts

```diff
@@ -95,3 +117,18 @@ function isTarget(device: Device): Predicate<Element> {
 function hasBoundingBox(device: Device): Predicate<Element> {
   return (element) => element.getBoundingBox(device).isSome();
 }
+
+function hasNonTargetText(device: Device): Predicate<Element> {
+  return (element) => {
+    if (isTarget(device)(element)) {
+      return false;
+    }
+
+    const children = element.children(Node.flatTree);
+    if (children.some(and(isText, isVisible(device)))) {
+      return true;
+    }
+
+    return children.filter(isElement).some(hasNonTargetText(device));
+  };
+}
```

Comment:
**Issue:** This may recurse deep in the tree in case of nested `<div>` that are typical of frameworks. It is probably OK-ish due to stopping at the first text, but would be better cached for performance.

Another possibility to avoid the double traversal of the tree is to handle it as part of `visit`. `visit` would build two maps: container -> targets, and container -> non-target. Then the applicability returns all values of the first map (targets) whose key is associated with an empty list in the second.